packages/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -4246,129 +4244,110 @@ export default class Meeting extends StatelessWebexPlugin {
-              LoggerProxy.logger.info(`${LOG_HEADER} PeerConnection CONNECTED`);
-
-              resolve(peerConnection);
-
-              return;
-            }
-            // Check if Peer Connection is STABLE (connected)
-            const stabilityTimeout = setTimeout(() => {
-              if (peerConnection.connectionState !== CONNECTION_STATE.CONNECTED) {
-                // TODO: Fix this after the error code pr goes in
-                reject(createMeetingsError(30202, 'Meeting connection failed'));
-              }
-              else {
-                LoggerProxy.logger.info(`${LOG_HEADER} PeerConnection CONNECTED`);
-                resolve(peerConnection);
-              }
-            }, PC_BAIL_TIMEOUT);
-
-            this.once(EVENT_TRIGGERS.MEDIA_READY, () => {
-              LoggerProxy.logger.info(`${LOG_HEADER} PeerConnection CONNECTED, clearing stability timer.`);
-              clearTimeout(stabilityTimeout);
-              resolve(peerConnection);
-            });
-          });
-        })
-        .then(() => {
-          if (mediaSettings && mediaSettings.sendShare && localShare) {
-            if (this.state === MEETING_STATE.STATES.JOINED) {
-              return this.share();
-            }
+      .then(
+        () => this.mediaProperties.waitForIceConnectedState()
+          .catch(() => {
+            throw createMeetingsError(30202, 'Meeting connection failed');
+          })
+      )
+      .then(() => {
+        LoggerProxy.logger.info(`${LOG_HEADER} PeerConnection CONNECTED`);
 
-            // When the self state changes to JOINED then request the floor
-            this.floorGrantPending = true;
+        if (mediaSettings && mediaSettings.sendShare && localShare) {
+          if (this.state === MEETING_STATE.STATES.JOINED) {
+            return this.share();
           }
 
-          Metrics.sendBehavioralMetric(
-            BEHAVIORAL_METRICS.ADD_MEDIA_SUCCESS,
-            {
-              correlation_id: this.correlationId,
-              locus_id: this.locusUrl.split('/').pop()
-            }
-          );
+          // When the self state changes to JOINED then request the floor
+          this.floorGrantPending = true;
+        }
 
-          return Promise.resolve();
-        }))
+        return {};
+      })
+      .then(() => this.mediaProperties.getCurrentConnectionType())
+      .then((connectionType) => {
+        Metrics.sendBehavioralMetric(
+          BEHAVIORAL_METRICS.ADD_MEDIA_SUCCESS,
+          {
+            correlation_id: this.correlationId,
+            locus_id: this.locusUrl.split('/').pop(),
+            connectionType
+          }
+        );
+      })
       .catch((error) => {
         // Clean up stats analyzer, peer connection, and turn off listeners
         const stopStatsAnalyzer = (this.statsAnalyzer) ? this.statsAnalyzer.stopAnalyzer() : Promise.resolve();
 
-        stopStatsAnalyzer
+        return stopStatsAnalyzer
```

Comment:
question: Prior to this change, the error caught on line 4346 would not have been re-thrown. Do we know what kind of errors these are?

Reply:
correct, previously the code was wrong - it was not throwing any error that happened during this long promise chain in addMedia(). I've found this out now when I added the "should reject if waitForIceConnectedState() rejects" test and it was not passing, because the error that is thrown in line 4318 in my new code (and line 4332 before my changes) was getting lost and not thrown to the client.

Reply:
it may be many types of errors, I don't know the full list, but whatever they are, they should all be thrown all the way up to the client